src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -435,4 +435,54 @@ unit_group_order_embedding.strict_mono
 
 end unit_group
 
+section maximal_ideal
+
+/-- The unique maximal ideal of a valuation subring, as a subsemigroup of `K`-/
```

Comment:
Ideal here means something different to `ideal`, right?

Reply:
Okay, I added a commit that might be preferable, using `nonunits` instead of `maximal_ideal` where confusion is possible.